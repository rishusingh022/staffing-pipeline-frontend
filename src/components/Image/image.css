.image-component {
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: inherit;
}

* {box-sizing: border-box}

.overlay-image-component{
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: inherit;
}
.container {
  position: relative;
  width: 100%;
  height: 100%;
  max-width: 300px;
}

.overlay {
  position: absolute;
  bottom: 0;
  background: rgb(0, 0, 0);
  background: rgba(0, 0, 0, 0.5); 
  color: #f1f1f1;
  width: 100%;
  height: 100%;
  transition: .5s ease;
  opacity:0;
  color: white;
  font-size: 70px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.container:hover .overlay {
  opacity: 1;
}