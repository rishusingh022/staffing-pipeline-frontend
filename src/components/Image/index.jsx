import React from 'react';
import './image.css';
import PropTypes from 'prop-types';

export default function Image({ imageUrl, altText }) {
  return <img src={imageUrl} alt={altText} className="image-component" />;
}

Image.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
};
