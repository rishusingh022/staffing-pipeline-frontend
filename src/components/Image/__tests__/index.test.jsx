import React from 'react';
import { render } from '@testing-library/react';
import Image from '..';

describe('Image Component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Image imageUrl={'http://surl.li/ffpzg'} altText="Google.com" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
