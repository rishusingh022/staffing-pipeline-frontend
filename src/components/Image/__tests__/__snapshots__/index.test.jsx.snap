// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render correctly 1`] = `
<DocumentFragment>
  <img
    alt="Google.com"
    class="image-component"
    src="http://surl.li/ffpzg"
  />
</DocumentFragment>
`;
